### PR TITLE
Refactor/More Help page & Public Charge Rule link config

### PR DESCRIPTION
**Pull request by msrezaie**
What (if anything) did you refactor?
The following changes makes it possible to have NC specific values for some of the left out hardcoded values found throughout the frontend code base:
- Refactored hardcoded values found in the More Help page to consume values coming from the config as well as created translations labels for some of the hardcoded text in the page.
- Refactored a hardcoded link found in the Step 2 of the screener which links to the [Colorado Department of Human Services Public Charge Rule](https://cdhs.colorado.gov/public-charge-rule-and-colorado-immigrants#:~:text=About%20public%20charge&text=The%20test%20looks%20at%20whether,affidavit%20of%20support%20or%20contract.) to get its value from the config.

Any other comments, questions, or concerns?
The following translation labels need to be added through the Translations API admin in order to change the text found in the More Help page:

1. - Label
    ```
    moreHelp.resource_name1
    ```
    - Default message:
    ```
    2-1-1 Colorado
    ```
2. - Label
    ```
    moreHelp.resource_phone1
    ```
    - Default message
    ```
    Dial 2-1-1 or 866.760.6489
    ```
3. - Label
    ```
    moreHelp.resource_name2
    ```
    - Default message
    ```
    Family Resource Center Association
    ```
4. - Label
    ```
    moreHelp.resource_description1
    ```
    - Default message
    ```
    Your local family resource center may be able to connect you to other resources and support services. Visit a center near you.
    ```

In order for the changes made in this PR to take effect, the following config objects needs to be added either manually through the Django admin, or through the configuration seed file `add_config.py`. The configuration seed file is updated in this [PR](https://github.com/Gary-Community-Ventures/benefits-api/pull/456):

1. - Name
    ```
    public_charge_rule
    ```
   - Data
   ```
   "{\"link\": \"https://cdhs.colorado.gov/public-charge-rule-and-colorado-immigrants#:~:text=About%20public%20charge&text=The%20test%20looks%20at%20whether,affidavit%20of%20support%20or%20contract.\"}"
   ```

2. - Name
    ```
    more_help_options
    ```
   - Data
   ```
   "{\"moreHelpOptions\": [{\"name\": {\"_default_message\": \"2-1-1 Colorado\", \"_label\": \"moreHelp.resource_name1\"}, \"link\": \"https://www.211colorado.org\", \"phone\": {\"_default_message\": \"Dial 2-1-1 or 866.760.6489\", \"_label\": \"moreHelp.resource_phone1\"}}, {\"name\": {\"_default_message\": \"Family Resource Center Association\", \"_label\": \"moreHelp.resource_name2\"}, \"description\": {\"_default_message\": \"Your local family resource center may be able to connect you to other resources and support services. Visit a center near you.\", \"_label\": \"moreHelp.resource_description1\"}, \"link\": \"https://maps.cofamilycenters.org\"}]}"
   ```

### CODE REVIEW

**Comment by CalebPena** [comment]
@msrezaie Could you change the base branch to `refactor/config-v2`

**Comment by msrezaie** [comment]
@CalebPena you mean the changes from this PR to go to `refactor/config-v2` instead of `dev`, or change the branch that `refactor/left-out-hardcoded-values` is based on to `refactor/config-v2` instead of `refactor/frontend-config-to-backend`?

**Comment by CalebPena** [comment]
> @CalebPena you mean the changes from this PR to go to `refactor/config-v2` instead of `dev`, or change the branch that `refactor/left-out-hardcoded-values` is based on to `refactor/config-v2` instead of `refactor/frontend-config-to-backend`?

Change the branch that `refactor/left-out-hardcoded-values` is based on to `refactor/config-v2` instead of `refactor/frontend-config-to-backend`

**Comment by msrezaie** [comment]
@CalebPena ok, I was thinking to get rid of this PR and the branch, and create a new branch and base it off of the `refactor/config-v2` and create a new PR from the new branch. Would this be ok, or do you recommend a better approach for doing this?

**Comment by CalebPena** [comment]
@msrezaie Feel free to create a new branch. You could also try rebasing this branch.

File: src/Components/LandingPage/LandingPage.tsx [hunk 1]
```diff
@@ -19,13 +19,13 @@ interface LandingPageProps {
...
 const LandingPage = ({ handleCheckboxChange }: LandingPageProps) => {
-  const { formData, locale, screenDoneLoading, theme } = useContext(Context);
+  const { formData, locale, screenDoneLoading, theme, config } = useContext(Context);
   const queryString = formData.immutableReferrer ? `?referrer=${formData.immutableReferrer}` : '';
   let { uuid } = useParams();
   const navigate = useNavigate();
   const privacyErrorController = useErrorController(termsOfServiceHasError, displayAgreeToTermsErrorMessage);
   const ageErrorController = useErrorController(termsOfServiceHasError, displayAgreeToTermsErrorMessage);
-
+  const { public_charge_rule } = config ?? {};
```
Comment (CalebPena) [inline src/Components/LandingPage/LandingPage.tsx:28]:
```suggestion
  const acuteConditionOptions = useConfig('public_charge_rule');
```

File: src/Components/MoreHelp/MoreHelp.tsx [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
+  name: JSX.Element;
+  description?: JSX.Element;
   link?: string;
-  phone?: string;
+  phone?: JSX.Element;
 };
 
 const MoreHelp = () => {
+  const { config } = useContext(Context);
+  const { more_help } = config ?? {};
```
Comment (CalebPena) [inline src/Components/MoreHelp/MoreHelp.tsx:15]:
```suggestion
  const acuteConditionOptions = useConfig('more_help');
```

File: src/Components/MoreHelp/MoreHelp.tsx [hunk 3]
```diff
@@ -1,33 +1,45 @@
...
+      name: <FormattedMessage id="moreHelp.resource_FRCA-name" defaultMessage="Family Resource Center Association" />,
+      description: (
+        <FormattedMessage
+          id="moreHelp.resource_FRCA-description"
+          defaultMessage="Your local family resource center may be able to connect you to other resources and support services. Visit a center near you."
+        />
+      ),
+      link: more_help?.resourceFRCALink,
     },
   ];
```
Comment (CalebPena) [inline src/Components/MoreHelp/MoreHelp.tsx:36]: This whole array should be added to the config.

**Review by CalebPena — CHANGES_REQUESTED** [review]
@mexi-cano Can you review please. I think that some things that were removed previously got added back in this pr.

@msrezaie Can you make a backend pr to update the `add_config` script?

File: src/Components/MoreHelp/MoreHelp.tsx [hunk 4]
```diff
@@ -1,45 +1,43 @@
...
   const displayResources = (resources: Resource[]) => {
-    return resources.map((resource) => {
+    return resources.map((resource, index) => {
       return (
-        <article key={resource.name} className="resource-card-article">
-          <h1 className="resource-header" key={resource.name}>
-            {resource.name}
+        <article key={index} className="resource-card-article">
+          <h1 className="resource-header" key={index}>
+            <FormattedMessage id={resource.name.props.id} defaultMessage={resource.name.props.defaultMessage} />
```
Comment (CalebPena) [inline src/Components/MoreHelp/MoreHelp.tsx:22]:
```suggestion
            {resource.name}
```
The config transforms it into a formatted message, so you can just inline it.

File: src/Components/MoreHelp/MoreHelp.tsx [hunk 5]
```diff
@@ -1,45 +1,43 @@
...
           </h1>
-          {resource.description && <p className="resource-desc">{resource.description}</p>}
+          {resource.description && (
+            <p className="resource-desc">
+              <FormattedMessage
+                id={resource.description.props.id}
+                defaultMessage={resource.description.props.defaultMessage}
+              />
```
Comment (CalebPena) [inline src/Components/MoreHelp/MoreHelp.tsx:29]:
```suggestion
              {resource.description}
```

File: src/Components/MoreHelp/MoreHelp.tsx [hunk 6]
```diff
@@ -1,45 +1,43 @@
...
             <a href={resource.link} className="resource-link" target="_blank">
               {resource.link}
             </a>
           )}
           {resource.phone && (
-            <a href={`tel:${resource.phone}`} className="resource-phone" target="_blank">
-              {resource.phone}
-            </a>
+            <p className="resource-phone">
+              <FormattedMessage id={resource.phone.props.id} defaultMessage={resource.phone.props.defaultMessage} />
```
Comment (CalebPena) [inline src/Components/MoreHelp/MoreHelp.tsx:39]:
```suggestion
              {resource.phone}
```

**Review by CalebPena — CHANGES_REQUESTED** [review]

**Review by CalebPena — APPROVED** [review]
Looks good.

**Pull request merged**